src/chunk/chunk_payload_data.rs

```diff
@@ -193,6 +193,9 @@ impl Chunk for ChunkPayloadData {
         if raw.len() < PAYLOAD_DATA_HEADER_SIZE {
             return Err(Error::ErrChunkPayloadSmall);
         }
+        if (header.value_length as usize) < PAYLOAD_DATA_HEADER_SIZE {
```

Comment:
Now that I think about it, most checks against raw.len() are redundant since the code needs to check that value_length is within raw.len() beforehand.

Reply:
Done